py_trees/decorators.py

```diff
@@ -196,15 +196,18 @@ def update(self):
         if the timeout is exceeded.
         """
         current_time = time.time()
-        if current_time > self.finish_time:
+        if self.decorated.status == common.Status.RUNNING and current_time > self.finish_time:
             self.feedback_message = "timed out"
             self.logger.debug("{}.update() {}".format(self.__class__.__name__, self.feedback_message))
             # invalidate the decorated (i.e. cancel it), could also put this logic in a terminate() method
             self.decorated.stop(common.Status.INVALID)
             return common.Status.FAILURE
-        # Don't show the time remaining, that will change the message every tick and make the tree hard to
-        # debug since it will record a continuous stream of events
-        self.feedback_message = self.decorated.feedback_message + " [timeout: {}]".format(self.finish_time)
+        if self.decorated.status == common.Status.RUNNING:
+            self.feedback_message = self.decorated.feedback_message + " [time remaining: {}s]".format(
```

Comment:
BTW I don't think we should have to mimic the child's feedback here since the child will be updating it and we can view it with print_ascii_tree.

Reply:
Yes, agreed! That too is a hangover from imposter days when that part of the feedback would have been masked.